msticpy/data/core/data_providers.py

```diff
@@ -449,22 +453,39 @@ def exec_query(self, query: str, **kwargs) -> Union[pd.DataFrame, Any]:
         """
         query_options = kwargs.pop("query_options", {}) or kwargs
         query_source = kwargs.pop("query_source", None)
-        result = self._query_provider.query(
-            query, query_source=query_source, **query_options
-        )
+
         if not self._additional_connections:
+            result = self._query_provider.query(
+               query, query_source=query_source, **query_options
+            )
             return result
+
         # run query against all connections
-        results = [result]
+        results = []
         print(f"Running query for {len(self._additional_connections)} connections.")
-        for con_name, connection in self._additional_connections.items():
-            print(f"{con_name}...")
-            try:
-                results.append(
-                    connection.query(query, query_source=query_source, **query_options)
-                )
-            except MsticpyDataQueryError:
-                print(f"Query {con_name} failed.")
+        if self._query_provider._support_async:
+            with ThreadPoolExecutor(max_workers=self._query_provider.max_threads) as executor:
```

Comment:
Not sure if this is the 'prettiest' way to do this since there are duplicate lines between the threaded variant and the sequential variant

Reply:
I think it's prob OK. I think it would more confusing to try to refactor out the common items.
One change I would make (I know it was my name originally) is change "MsticpyInstance" to "DataProviderInstance"